matlab/GLMestimatesingletrial.m

```diff
@@ -1,123 +1,145 @@
 function results = GLMestimatesingletrial(design,data,stimdur,tr,outputdir,opt)
-
-% function results = GLMestimatesingletrial(design,data,stimdur,tr,outputdir,opt)
+%
+% USAGE::
+%
+%   results = GLMestimatesingletrial(design,data,stimdur,tr,outputdir,opt)
 %
 % <design> is the experimental design. There are two possible cases:
+%
 %   1. A where A is a matrix with dimensions time x conditions.
 %      Each column should be zeros except for ones indicating condition onsets.
 %   2. {A1 A2 A3 ... An} where each of the A's are like the previous case.
 %      The different A's correspond to different runs, and different runs
 %      can have different numbers of time points. However, all A's must have
 %      the same number of conditions.
-%   Note that we ultimately compute single-trial response estimates (one estimate for 
-%   each condition onset), and these will be provided in chronological order. However,
-%   by specifying that a given condition occurs more than one time over the course
-%   of the experiment, this information can and will be used for cross-validation purposes.
+%
+% Note that we ultimately compute single-trial response estimates (one estimate for 
+% each condition onset), and these will be provided in chronological order. However,
+% by specifying that a given condition occurs more than one time over the course
+% of the experiment, this information can and will be used for cross-validation purposes.
+%
```

Comment:
I did a "quick fix" of the help section of the main matlab function so that it renders correctly

https://glmsingle.readthedocs.io/en/latest/matlab.html

The only change you would have to do to your usually commenting style so that it plays nice with the restrcutured text markup to sphinx expects would be to use lineskip rather than indents for each of the your parameter definition paragraph.

I will still send some typical template function in case you wanted to adopt something that ressembles some of the other typical commenting styles used in sphinx.

Reply:
In case you want to browse the whole function directly: see it on my fork

https://github.com/Remi-Gau/GLMsingle/blob/add_sphinx_doc/matlab/GLMestimatesingletrial.m